src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/callback/ExceptionRecordValidator.java

```diff
@@ -136,6 +142,18 @@ public ExceptionRecordValidator() {
         ).toValidation().mapError(throwable -> "Invalid scannedDocuments format. Error: " + throwable.getMessage());
     }
 
+    private String getEnvelopeId(CaseDetails caseDetails) {
+        return Optional.ofNullable(caseDetails)
+            .map(CaseDetails::getData)
+            .map(data -> (String) data.get(ENVELOPE_ID))
+            .orElseThrow(
+                // we can't apply standard validation, as the caseworker doesn't even know about envelope ID field
+                () -> new UnprocessableCaseDataException(
```

Comment:
do we have a test to cover this ?

Reply:
We do, we have tests for verifying that the callback endpoint will return a 422 response with the right body when envelope ID is missing. Earlier today I also created a ticket for `ExceptionRecordValidator` unit tests (https://tools.hmcts.net/jira/browse/BPS-1336), as they're all missing.